bindings/dart/lib/src/common/reranker/data_provider.dart

```diff
@@ -3,16 +3,55 @@ import 'package:hex/hex.dart' show HEX;
 
 part 'assets.dart';
 
+/// An asset consists of an URL suffix, a [`Checksum`] and optionally
+/// a list of [`Fragment`]s.
+///
+/// The base URL (defined by the caller) concatenated with the URL suffix
+/// creates the URL to fetch an asset.
+///
+/// The checksum is the hash of an asset and can be used to verify its
+/// integrity after it has been fetched.
+///
+/// In order to keep larger assets in the http cache of a browser,
+/// an asset might be split into multiple fragments.
+///
+/// Implementation details for fetching assets:
+///
+/// If the list of fragments is empty, the caller may use the URL suffix of the
```

Comment:
I would say that he `must` use the url of the asset because he has no other choise